### PR TITLE
Feat/color picker

### PR DESCRIPTION
**Pull request by Mark-Krensel**
please check my new feature: the diaper color-picker!
it is accessible from the the big + icon on the bottom and then the diaper icon

### CODE REVIEW

File: components/Card.js [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
-    const m = Math.floor((timeStamp % (1000 * 60 * 60)) / (1000 * 60));
-    const time = ` @${h}:${m}h`;
-    return time;
-  }
+  // function calcTime(timeStamp) {
+  //   const h = Math.floor((timeStamp % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+  //   const m = Math.floor((timeStamp % (1000 * 60 * 60)) / (1000 * 60));
+  //   const time = ` @${h}:${m}h`;
+  //   return time;
+  // }
```
Comment (kai-franke) [inline components/Card.js:17]: Can this comment be deleted?

File: components/Card.js [hunk 2]
```diff
@@ -95,11 +106,37 @@ export default function Card({ date, id, weights, handleDelete, heights, feastTi
...
+      {diaperColors.length !== 0 && (
+        <>
+          <AttributeText>Diaper Color</AttributeText>
+          <AttributeColorList>
+            {diaperColors.map((diaperColor) => (
+              <AttributeColorListItem key={diaperColor._id}>
+                {pathname === '/' && (
+                  <DeleteSingleColor
+                    inputColor={diaperColor.value}
+                    aria-label="delete single data point"
```
Comment (kai-franke) [inline components/Card.js]: Is this aria label really meaningful? If I understand the use correctly, for me this should say "Delete single color".

File: components/CardContainer.js [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+  position: relative;
   display: flex;
   gap: 1em;
   flex-wrap: wrap;
   justify-content: center;
   padding: 1em;
-  margin-top: 3.5em;
-  max-height: calc(100vh - 3.5em);
+  margin: 3.5em auto 0 auto;
+  /* max-height: calc(100vh - 3.5em); */
```
Comment (kai-franke) [inline components/CardContainer.js]: Can this comment be removed?

File: components/Dashboard.js [hunk 4]
```diff
@@ -91,14 +92,14 @@ const DashboardCard = styled.article`
...
-  border: 1px solid var(--text-primary);
+  border: 2px solid var(--text-primary);
   color: var(--text-primary);
   border-radius: 1em;
   box-shadow: var(--shadow-elevation);
   height: auto;
   max-height: 12.5em;
   width: 95%;
-  max-width: 32em;
+  max-width: 500px
```
Comment (kai-franke) [inline components/Dashboard.js]: Just because I'm curious: Why did you switch to an absolute value?

File: components/SrOnly.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import styled from 'styled-components';
+
+const SrOnly = styled.span`
+  clip: rect(1px, 1px, 1px, 1px);
```
Comment (kai-franke) [inline components/SrOnly.js:4]: MDN says that `clip` is deprecated and should no longer be used.
https://developer.mozilla.org/en-US/docs/Web/CSS/clip?retiredLocale=de

File: components/Card.js [hunk 6]
```diff
@@ -95,11 +106,37 @@ export default function Card({ date, id, weights, handleDelete, heights, feastTi
...
+                    onClick={() => handleDelete(id, diaperColor._id, 'diaperColors')}
+                  >
+                    <X fontSize="1.5em" alt="delete" />
+                  </DeleteSingleColor>
+                )}
+                <ColorRoundCase inputColor={diaperColor.value}>
+                  {/* ToDo swap hex number for color name */}
+                  <SrOnly>{diaperColor.value}</SrOnly>
```
Comment (kai-franke) [inline components/Card.js]: What is this component for? Is this for screen readers? I'm not sure if it really needs a component for that usecase. Maybe you can just use
`ariaLabel={diaperColor.value}` or `aria-label={diaperColor.value}`
on the ColorRoundCase component?

File: components/forms/FormDB.js [hunk 7]
```diff
@@ -39,18 +41,24 @@ export default function Form({ onSubmit, setShownAttribute, shownAttribute }) {
     } else if (shownAttribute === 'feastTime' && foodSourceInput == '') {
       alert('On which side did you feed the baby, or did you use a bottle?');
       return false;
+    } else if (shownAttribute === 'diaper' && diaperColorInput == '') {
+      alert('With which diaper substance color have you been surprised with?');
```
Comment (kai-franke) [inline components/forms/FormDB.js:45]: 😆

File: components/forms/FormDB.js [hunk 8]
```diff
@@ -84,6 +92,19 @@ export default function Form({ onSubmit, setShownAttribute, shownAttribute }) {
           step="0.1"
         />
       )}
+      {shownAttribute === 'diaper' && (
+        <NakedFieldset aria-label="selection of diaper contet colors">
```
Comment (kai-franke) [inline components/forms/FormDB.js]: typo: content

File: components/icons/Clock.js [hunk 9]
```diff
@@ -1,15 +1,16 @@
...
-      enableBackground: "new 0 0 24 24",
+      enableBackground: 'new 0 0 24 24',
     }}
     viewBox="0 0 24 24"
     xmlSpace="preserve"
     xmlns="http://www.w3.org/2000/svg"
     width="1em"
     height="1em"
     fill="currentColor"
+    // stroke="black"
```
Comment (kai-franke) [inline components/icons/Clock.js:13]: Can this comment be removed?

**Review by kai-franke — COMMENTED** [review]
Great feature, great work! :)
React was easy to read but I'm not really into the MongoDB thing anymore. Maybe you should get a second opinion on that topic.

File: components/Dashboard.js (hunk 4 above)
Comment (Mark-Krensel) [reply components/Dashboard.js]: it's a matter of styling. I never want the dashboard to be wider than a  specific, fixed number of pixels. it shouldn't change even if you increase or decrease the font-size.

File: components/SrOnly.js (hunk 5 above)
Comment (Mark-Krensel) [reply components/SrOnly.js:4]: true, however a topic for another time ;)

File: components/Card.js (hunk 6 above)
Comment (Mark-Krensel) [reply components/Card.js]: this is a really good idea!

File: components/Card.js (hunk 1 above)
Comment (Mark-Krensel) [reply components/Card.js:17]: unfortunatly no. need that code snipped for experimentation :/

**Pull request merged**